lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -264,26 +277,22 @@ def main():
     path = module.params.get('path')
     new_name = module.params.get('new_name')
     new_path = module.params.get('new_path')
-    cert_chain = module.params.get('cert_chain')
     dup_ok = module.params.get('dup_ok')
-    if state == 'present':
-        if module.params.get('cert') is not None:
-            cert = module.params.get('cert')
-        if module.params.get('key') is not None:
-            key = module.params.get('key')
-        if module.params.get('cert_chain') is not None:
-            cert_chain = module.params.get('cert_chain')
+    if state == 'present' and not new_name and not new_path:
+        cert = module.params.get('cert')
+        key = module.params.get('key')
+        cert_chain = module.params.get('cert_chain')
     else:
-        key=cert=chain=None
+        cert = key = cert_chain = None
 
-    orig_certs = [ctb['server_certificate_name'] for ctb in \
+    orig_cert_names = [ctb['server_certificate_name'] for ctb in \
                                                     iam.get_all_server_certs().\
                                                     list_server_certificates_result.\
                                                     server_certificate_metadata_list]
-    orig_bodies = [iam.get_server_certificate(thing).\
-                  get_server_certificate_result.\
-                  certificate_body \
-                  for thing in orig_certs]
+    orig_cert_bodies = [iam.get_server_certificate(thing).\
+                        get_server_certificate_result.\
+                        certificate_body \
```

Comment:
Lol.  I actually changed some of this stuff but didn't push it since it was already there because and I thought less change was better.